backend_new/src/services/reserved-community-type.service.ts

```diff
@@ -0,0 +1,140 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { Prisma } from '@prisma/client';
+import { ReservedCommunityType } from '../dtos/reserved-community-types/reserved-community-type-get.dto';
+import { ReservedCommunitTypeCreate } from '../dtos/reserved-community-types/reserved-community-type-create.dto';
+import { mapTo } from '../utilities/mapTo';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { ReservedCommunityTypeQueryParams } from '../dtos/reserved-community-types/reserved-community-type-query-params.dto';
+
+/*
+  this is the service for reserved community types
+  it handles all the backend's business logic for reading/writing/deleting reserved community type data
+*/
+
+const view: Prisma.ReservedCommunityTypesInclude = {
+  jurisdictions: true,
+};
+
+@Injectable()
+export class ReservedCommunityTypeService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of reserved community types given the params passed in
+  */
+  async list(params: ReservedCommunityTypeQueryParams) {
+    const rawReservedCommunityTypes =
+      await this.prisma.reservedCommunityTypes.findMany({
+        include: view,
+        where: this.buildWhereClause(params),
+      });
+    return mapTo(ReservedCommunityType, rawReservedCommunityTypes);
+  }
```

Comment:
Like you did with ami charts, please add in the return types to the methods here.

Reply:
⚡